src/HealthGPS/hierarchical_model_registry.h

```diff
@@ -35,4 +39,15 @@ namespace hgps {
 
 		return registry;
 	}
+
+	/// @brief Gets the default production instance builders for energy balance models
+	/// @return The energy balance models builder functions registry
+	std::map<HierarchicalModelType, detail::EnergyBalanceModelBuilder> get_default_energy_balance_model_registry() {
+		auto registry = std::map<HierarchicalModelType, detail::EnergyBalanceModelBuilder>{
+			{HierarchicalModelType::Dynamic, [](EnergyBalanceModelDefinition& definition) {
+				return std::make_unique<EnergyBalanceModel>(definition); }},
+		};
+
+		return registry;
+	}
```

Comment:
All this ``*ModelBuilder`` stuff is something that we should be able to factor out at some point... This is a particularly cumbersome design :disappointed:

Reply:
Agreed. in fact this whole nasty file can go with the new changes.